system/jlib/jstats.cpp

```diff
@@ -1086,39 +1086,21 @@ static int compareUnsigned(unsigned const * left, unsigned const * right)
     return (*left < *right) ? -1 : (*left > *right) ? +1 : 0;
 }
 
-StatisticsMapping::StatisticsMapping(StatisticKind kind, ...)
+StatisticsMapping::StatisticsMapping(const std::initializer_list<StatisticKind> &kinds)
 {
-    if (kind != StKindNone)
-    {
-        indexToKind.append(kind);
-        va_list args;
-        va_start(args, kind);
-        for (;;)
-        {
-            unsigned next  = va_arg(args, unsigned);
-            if (!next)
-                break;
-            indexToKind.appendUniq(next);
-        }
-        va_end(args);
-    }
+    for (auto kind : kinds)
+        if (kind != StKindNone)
+            indexToKind.append(kind);
```

Comment:
Should also be appendUnique.  Possible would be better too assert that it isn't StKindNone.